src/open_inwoner/configurations/bootstrap/models.py

```diff
@@ -0,0 +1,296 @@
+import dataclasses
+from dataclasses import dataclass, field
+from typing import Any, Iterator, TypeAlias
+
+from django.contrib.postgres.fields import ArrayField
+from django.db.models.fields import NOT_PROVIDED
+from django.db.models.fields.json import JSONField
+from django.db.models.fields.related import ForeignKey, OneToOneField
+
+from digid_eherkenning.models import DigidConfiguration
+
+from digid_eherkenning_oidc_generics.models import (
+    OpenIDConnectDigiDConfig,
+    OpenIDConnectEHerkenningConfig,
+)
+from open_inwoner.configurations.models import SiteConfiguration
+from open_inwoner.openklant.models import OpenKlantConfig
+from open_inwoner.openzaak.models import OpenZaakConfig
+
+from .dataclasses import ConfigField, Fields
+
+ConfigModel: TypeAlias = (
+    SiteConfiguration
+    | OpenKlantConfig
+    | OpenZaakConfig
+    | OpenIDConnectDigiDConfig
+    | OpenIDConnectEHerkenningConfig
+)
+
+
+class ConfigSettingsBase:
+    model: ConfigModel
+    display_name: str
+    namespace: str
+    required_fields: tuple[str, ...]
+    included_fields: tuple[str, ...]
+    excluded_fields: tuple[str, ...]
+
+    def __init__(self):
+        self.config_fields = Fields(all=set(), required=set())
+
+        self.create_model_config_fields(
+            require=self.required_fields,
+            exclude=self.excluded_fields,
+            include=self.included_fields,
+            model=self.model,
+        )
+
+    @classmethod
+    def get_setting_name(cls, field: ConfigField) -> str:
+        return f"{cls.namespace}_" + field.name.upper()
+
+    @staticmethod
+    def get_default_value(field: Any) -> str:
+        default = field.default
+
+        if default is NOT_PROVIDED:
+            return "No default"
+        if callable(default):
+            default = default.__call__()
```

Comment:
Sometimes I wonder how you come up with stuff.. like I have never seen a callable being called with `__call__()` :confused: 

Let's just call it:

````python
    if callable(default):
        default = default()
````

Anyway, as a base rule it is a big smell if you need dunder methods for general tasks, so if you feel you need to then reevaluate if you're on the right track.